### ISSUE TITLE
luaL_typeerror

### ISSUE THREAD

**Issue by devurandom** [keplerproject/lua-compat-5.3#46]
`luaL_typeerror` is not an officially documented function, but it is exported by Lua 5.3 and the code is simple:
```c
int luaL_typeerror (lua_State *L, int arg, const char *tname) {
  const char *msg;
  const char *typearg;  /* name for the type of the actual argument */
  if (luaL_getmetafield(L, arg, "__name") == LUA_TSTRING)
    typearg = lua_tostring(L, -1);  /* use the given type name */
  else if (lua_type(L, arg) == LUA_TLIGHTUSERDATA)
    typearg = "light userdata";  /* special name for messages */
  else
    typearg = luaL_typename(L, arg);  /* standard name */
  msg = lua_pushfstring(L, "%s expected, got %s", tname, typearg);
  return luaL_argerror(L, arg, msg);
}
```

**Comment by siffiejoe** [comment]
I could not find a `luaL_typeerror` function in the Lua 5.3.5 sources. There is only the static `typeerror` helper function in `lauxlib.c`: https://www.lua.org/source/5.3/lauxlib.c.html#typeerror

**Comment by devurandom** [comment]
I found the declaration here: https://github.com/lua/lua/blob/master/lauxlib.h#L51

**Comment by daurnimator** [comment]
> I found the declaration here: https://github.com/lua/lua/blob/master/lauxlib.h#L51

You're looking at master which is the un-released lua 5.4.
`luaL_typeerror` was added in https://github.com/lua/lua/commit/46beca5bed8a7700b18100fe48a78373be5055f9

**Comment by devurandom** [comment]
Oh, I did not even know Lua development happened in the open now. I assumed that repository just tracked the released versions.
